_posts/2019-03-04-jumping-on-the-dash-less-bandwagon.markdown

```diff
@@ -0,0 +1,64 @@
+
+Firstly, it removes the infrastructure downtime between tasks. In the past you’d have to wait for servers to become available. Then you need to wait for software (like application servers) to be installed and configured. Then it was gaining database access for your code. Etcetera etcetera etcetera. With serverless, once you have your access permissions, you can have a brand-new application, in production, within hours. Everything needed to host and operate your application is provided to you by the cloud provider.
+
+Next, with all this infrastructure provided to you, gone are hours or days of writing boilerplate code. All the lambda functions I’ve written listen for an incoming event (http, file, database, whatever) and consume it. No setting up scaffolding code. No writing complex convoluted setup logic for unit tests. Just the minimum code to consume the event and suite of testing scenarios to verify the behaviours that I expect. [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) at its finest.
+
+Time to think in terms of events and states as ultimately you’re responding to state change events within the bounds of a fully distributed system. This requires a more disciplined focus on what triggers these events and how things can, and will, [go awry at any time](https://en.wikipedia.org/wiki/Chaos_engineering).
+
+The architecture diagram below for a photo management service, on a recent project, demonstrates just how powerful Serverless can be. The service allows end-users to take, upload and manage maintenance photos of the organisation's physical equipment in the field. To deliver this, nothing more beyond CloudFormation scripts and Lambda code was needed.
+
+<img style="display: block; margin-left: auto; margin-right: auto;" alt="photo management service" src="{{ site.url }}/img/bandwagonless/photo-app.png"/>
+
+**Link of Interest:** [6 things I’ve learned in my first 6 months using serverless](https://read.acloud.guru/six-months-of-serverless-lessons-learned-f6da86a73526)
+
+## Wrapping Up
+
+For years now the phrase “start-ups are coming to eat your lunch” has been shouted from all corners of the Internet. This warning hasn’t changed, but what has is the extremely low barrier of entry to eat it. A few years ago, start-ups were still paying for virtual worlds that mirrored the physical; EC2 servers, VPC networking, relational databases designed just like on premise, VPN links back to data centres, etc. Dedicated people were also required (and paid) to build, manage and operate all these things.
+
+Today with serverless, entire development-to-production environments can be scripted and deployed by a single developer. The literal cost? A few cents at most. In fact, your time spent reading this post has probably cost more to the company you work for than the yearly operating costs of the serverless systems I’ve worked on.
```

Comment:
You could probably call out that their are entire businesses that are now built on serverless - acloudguru. Similarly, Fender (the guitar company) has gone entirely serverless (Fender Play, Their website, and mobile apps) now as well I believe - if you need an example of a 70 year old established company doing it.  I'm sure there are more.